### PR TITLE
Singlevisit documentation update

### PR DESCRIPTION
**Pull request by dulude**
**Associated JIRA tickets**
- [HLA-305](https://jira.stsci.edu/browse/HLA-305)

**High-level Summary**
- This PR contains a number of updates/changes/tweaks to the singlevisit.rst documentation file.
- There were also updates/changes/tweaks to several other .rst files, as well as docstrings in a couple of python scripts. These were both in support of the singlevisit.rst updates, as well as stand-alone fixes to other documentation issues that I encountered.
- There is also the initial commit of sourcelist_flag_breakdown.py, a code which simple gives the flag break down for a single user-specified .ecsv catalog file.

**Specific Changes**
- singlevisit.rst:
  - Resolved most of @mdlpstsci 's suggestions in her [singlevisit.rst comment](https://jira.stsci.edu/browse/HLA-305?focusedCommentId=495140&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-495140) from Ticket HLA-305.
- catalog_generation.rst:
  - added "Attention" item to the end of section 2.4.2 that informs users about the fact that the final filter catalogs only contain the "best" sources
- runastrodriz.rst:
  - fixed syntax issue in the "Update the WCS" section
- runsinglehap.rst:
  - Added a few lines of code in order to show generate_custom_svm_param_file documentation
- drizzlepac/devutils/sourcelist_flag_breakdown.py :
  - Initial commit.
- drizzlepac/hapsequencer.py:
  - added two log info statements to create_catalog_products() informing users how many catalog rows with flag values > flag_trim_value were removed from final output catalogs 
- drizzlepac/haputils/generate_custom_svm_param_file.py:
  - Brought docstrings up to spec
- wfc3_ir_catalog_generation_all.json: 
  - changed flag_trim_value value from 65368 to 5

### CODE REVIEW

File: doc/source/catalog_generation.rst [hunk 1]
```diff
@@ -286,6 +288,15 @@ these values. Specific flag values are defined below in table 2:
     |            | or other region with a low number of input images         |
     +------------+-----------------------------------------------------------+
 
+.. attention::
+
+    The final output filter-specific sourcelists do not contain all detected sources. Sources that are considered
+    scientifically dubious are filtered out and not written to the final source catalogs. For all detectors other than
+    WFC3/IR, sources with a flag value greater than 5 are filtered out. For WFC3/IR, sources with a flag value of
+    65368 are filtered out. Users can adjust this value using a custom input parameter file and changing the
```
Comment (mdlpstsci) [inline doc/source/catalog_generation.rst:296]: Maybe the flag value got changed in the code, but it also needs to be changed to "5" here.

**Review by mdlpstsci — CHANGES_REQUESTED** [review]
Just a few minor changes.  It is important to get this update merged as there are yet more changes being done by other team members describing specialized updates.

File: doc/source/singlevisit.rst [hunk 2]
```diff
@@ -80,16 +81,20 @@ convention used for the names of these input and output files uses these
...
   * **<propid>** : the proposal ID for this visit
-  * **<obsetid>** : the 2-digit visit ID from this proposal
-  * **<instr>** : 3 letter designation of the instrument used for the observations
+  * **<obsetid>** : the 2-digit visit ID from this proposal 
+  * **<instr>** : 3 or 4 letter designation of the instrument used for the observations
   * **<detector>** : name of the detector used for the observations
   * **<filter>** : hyphen-separated list of filter names used for the observations
   * **<ipppssoo>** : standard 8 character rootname for a single exposure defined by the pipeline
   * **<ipppss>** : standard 6 character designation of the **<instr>**/**<propid>**/**<obsetid>** for this visit
+  * **asn.fits** : suffix for the association table
```
[the remark this answers is not in the document]
Comment (dulude) [reply doc/source/singlevisit.rst]: issue resolved by commit [0f93085](https://github.com/spacetelescope/drizzlepac/pull/947/commits/0f930854c3d1ead3966a8b999bf7a94a2d8c7426)

File: doc/source/singlevisit.rst [hunk 3]
```diff
@@ -55,20 +55,21 @@ relies on the results of the standard astrometric
...
-
-  * Define what output products can be generated
-  * Align all exposures in a relative sense (all to each other)
-  * Create a composite source catalog from all aligned input exposures
-  * Cross-match and fit this composite catalog to GAIA to determine new WCS solution
-  * Update renamed input exposures with results of alignment to GAIA
-  * Create each of the output products using the updated WCS solutions
-
+  #. interpret the list of filenames for all exposures taken as part of a single visit
+  #. copy the pipeline-calibrated (FLT/FLC) files to the current directory for processing
```
[the remark this answers is not in the document]
Comment (dulude) [reply doc/source/singlevisit.rst:60]: Issue resolved by commit [cb9f38e](https://github.com/spacetelescope/drizzlepac/pull/947/commits/cb9f38e913d1d312cb750cb720f5d48a95134c41)

**Review by mdlpstsci — APPROVED** [review]

**Review by mdlpstsci — APPROVED** [review]
All issues addressed.

**Pull request merged**